x/conflict/types/relay_finalization_metadata.go

```diff
@@ -1,38 +1,72 @@
 package types
 
 import (
+	"encoding/json"
+	fmt "fmt"
+
 	tendermintcrypto "github.com/cometbft/cometbft/crypto"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
 	"github.com/lavanet/lava/utils/sigs"
 	pairingtypes "github.com/lavanet/lava/x/pairing/types"
 )
 
-type RelayFinalizationMetaData struct {
-	MetaData ReplyMetadata
-	Request  pairingtypes.RelayRequest
-	Addr     sdk.AccAddress
+func NewRelayFinalizationMetaDataFromReplyMetadataAndRelayRequest(reply ReplyMetadata, req pairingtypes.RelayRequest, consumerAddr sdk.AccAddress) RelayFinalization {
+	return RelayFinalization{
+		FinalizedBlocksHashes: reply.FinalizedBlocksHashes,
+		LatestBlock:           reply.LatestBlock,
+		ConsumerAddress:       consumerAddr.String(),
+		RelaySession:          req.RelaySession,
+		SigBlocks:             reply.SigBlocks,
+	}
 }
 
-func NewRelayFinalizationMetaData(meta ReplyMetadata, req pairingtypes.RelayRequest, addr sdk.AccAddress) RelayFinalizationMetaData {
-	return RelayFinalizationMetaData{MetaData: meta, Request: req, Addr: addr}
+func NewRelayFinalizationMetaDataFromRelaySessionAndRelayReply(relaySession *pairingtypes.RelaySession, relayReply *pairingtypes.RelayReply, consumerAddr sdk.AccAddress) RelayFinalization {
+	return RelayFinalization{
+		FinalizedBlocksHashes: relayReply.FinalizedBlocksHashes,
+		LatestBlock:           relayReply.LatestBlock,
+		ConsumerAddress:       consumerAddr.String(),
+		RelaySession:          relaySession,
+		SigBlocks:             relayReply.SigBlocks,
+	}
 }
 
-func (rfm RelayFinalizationMetaData) GetSignature() []byte {
-	return rfm.MetaData.SigBlocks
+func (rfm RelayFinalization) GetSignature() []byte {
+	return rfm.SigBlocks
 }
 
-func (rfm RelayFinalizationMetaData) DataToSign() []byte {
-	relaySessionHash := tendermintcrypto.Sha256(rfm.Request.RelaySession.CalculateHashForFinalization())
-	latestBlockBytes := sigs.EncodeUint64(uint64(rfm.MetaData.LatestBlock))
+func (rf RelayFinalization) DataToSign() []byte {
+	if rf.RelaySession == nil {
+		return nil
+	}
+
+	relaySessionHash := tendermintcrypto.Sha256(rf.RelaySession.CalculateHashForFinalization())
+	latestBlockBytes := sigs.EncodeUint64(uint64(rf.LatestBlock))
 	msgParts := [][]byte{
 		latestBlockBytes,
-		rfm.MetaData.FinalizedBlocksHashes,
-		rfm.Addr,
+		rf.FinalizedBlocksHashes,
+		[]byte(rf.ConsumerAddress),
 		relaySessionHash,
 	}
 	return sigs.Join(msgParts)
 }
 
-func (rfm RelayFinalizationMetaData) HashRounds() int {
+func (rfm RelayFinalization) HashRounds() int {
 	return 1
```

Comment:
I think it's actually more clear this way. This is kind of constant, since this is the only place it is written like that. If you want to know how many hash rounds you do for a `signable` struct, you always got to its `HashRounds` function